doc/Data Plane Code Generation/ServiceDrivenEvolution.md

```diff
@@ -0,0 +1,179 @@
+{
+    ...
+    try
+    {
+        using HttpMessage message = CreatePostParametersRequest(content, contentType, context);
+        return await _pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
+    }
+    ...
+}
+
+internal HttpMessage CreatePostParametersRequest(RequestContent content, ContentType contentType, RequestContext context)
+{
+    ...
+    request.Headers.Add("Content-Type", contentType.ToString());
+    request.Content = content;
+    return message;
+}
+```
+
+Since the V2 client added a required parameter to an existing client method, it breaks the code that is written for the V1 client. We can manually add an overload method that has the same signature as V1 to make it backward compatible:
+
+``` C#
+[EditorBrowsable(EditorBrowsableState.Never)]
+public virtual async Task<Response> PostParametersAsync(RequestContent content, RequestContext context = null)
+{
+    ...
+    try
+    {
+        using HttpMessage message = CreatePostParametersRequest(content, ContentType.ApplicationJson, context);
+        return await _pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
+    }
+    ...
+}
+```
+
+**Notice:**
+1. We leverage the `CreatePostParametersRequest` method in V2 but always pass in the `ContentType.ApplicationJson` as the `contentType` parameter in the manual method to make the behavior align with V1.
+2. Whether to add the attribute `[EditorBrowsable(EditorBrowsableState.Never)]` depends on whether you want to discourage callers from using this. E.g., if service add a new ContentType value to improve performance, you should add this attribute so new users wouldn't discover it, while if service has default preference of old ContentType value, you should not add this attribute to make it easy for users to discover the default.
+
+## A method changes a required parameter to optional
+
+Generated code in a V1 client with two required parameters:
+
+``` C#
+public virtual async Task<Response> PutRequiredOptionalAsync(string param1, string param2, RequestContext context = null)
+{
+    ...
+    try
+    {
+        using HttpMessage message = CreatePutRequiredOptionalRequest(param1, param2, context);
+        return await _pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
+    }
+    ...
+}
+```
+
+**Required input is changed to an optional input when it is in the last position (i.e., `param2`).**
+
+This case is safe to generate as:
+
+``` C#
+public virtual async Task<Response> PutRequiredOptionalAsync(string param1, string param2 = null, RequestContext context = null)
+{
+    ...
+    try
+    {
+        using HttpMessage message = CreatePutRequiredOptionalRequest(param1, param2, context);
+        return await _pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
+    }
+    ...
+}
+```
+
+**Required input is changed to an optional input when it is not in the last position (e.g., `param1`).**
```

Comment:
It feels like this section would go better after the first section -- parameter ordering for these cases is closely related.